src/views/templates/actions/components/SelectSource.tsx

```diff
@@ -178,7 +178,11 @@ export const SelectSource: React.FC<SelectSourceProps> = ({
       )}
 
       {withSize && selectedSourceType !== SOURCE_TYPES.defaultSource && (
-        <VolumeSize quantity={volumeQuantity} onChange={setVolumeQuantity} />
+        <DiskSizeNumberInput
```

Comment:
I think DiskSizeNumberInput can be moved to more general location instead of nested under disk modal, also when extracting can be rename to more general name such as SizeUnitsNumberInput, same goes to prop diskSize = size

Reply:
Done. I went with `MemorySizeInput` since RAM and storage are both types of memory.